src/main/java/org/jenkinsci/plugins/puppet/track/RootActionImpl.java

```
@@ -36,13 +38,18 @@ public String getUrlName() {
     @RequirePOST
     public HttpResponse doReport(StaplerRequest req) throws IOException {
         // TODO: stapler YAML support
-
-        PuppetReport.load(req.getReader()).process();
-
+        LOGGER.log(Level.FINE,"Puppet plugin has received a request from " + req.getRemoteAddr());
+        try {
+            PuppetReport.load(req.getReader()).process();
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE,"Puppet plugin couldn't process the report sent by " + req.getRemoteAddr());
```

Log the exception, and throw it back or this SERVERE error will be silently ignored.